assets/scripts/segments/Segment.jsx

```diff
@@ -308,12 +314,18 @@ function mapStateToProps (state) {
     locale: state.locale.locale,
     infoBubbleHovered: state.infoBubble.mouseInside,
     descriptionVisible: state.infoBubble.descriptionVisible,
-    activeSegment: (typeof state.ui.activeSegment === 'number') ? state.ui.activeSegment : null
+    activeSegment: (typeof state.ui.activeSegment === 'number') ? state.ui.activeSegment : null,
+    resolution: state.ui.unitSettings.resolution
   }
 }
 
-const mapDispatchToProps = {
-  setActiveSegment
+function mapDispatchToProps (dispatch, ownProps) {
+  return {
+    setActiveSegment: (position) => { dispatch(setActiveSegment(position)) },
+    removeSegment: (position) => { dispatch(removeSegmentAction(position)) },
+    clearSegments: () => { dispatch(clearSegments()) },
+    incrementSegmentWidth: (dataNo, add, precise, resizeType) => dispatch(incrementSegmentWidth(dataNo, add, precise, ownProps.actualWidth, resizeType))
+  }
```

Comment:
Are we not able to use the object form of `mapDispatchToProps` here?

Reply:
not really, since in `incrementSegmentWidth` we're using `ownProps` to set the `actualWidth` 
we could do that somewhere else, but I thought this is a great place to remove duplication and this way it's less code.

Reply:
Gotcha - let's leave this alone